### PR TITLE
fix(Communities): Introduce request id for checking community permissions

### LINKED ISSUE #14376: Chat content updates in the background while playing with the Edit Shared Popup addresses
# Bug Report

## Description

You can see it happening in the video from this comment: https://github.com/status-im/status-go/pull/5023#issuecomment-2043655144

Basically, if you have a community with permissioned channels and have different accounts that have different tokens, if you go in a channel that has permissions, then go in the Edit Shared Addresses popup, when you select/unselect accounts, the permission check result will not only affect the popup, but also the chat in the background.

One solution would be to add a UUID to the permission check, so that when we send the result from the service, only the module that asked for that check handles the result. In this case, only the communities module should react to that event.

## Steps to reproduce

0. Have a community with some channels containing permissions to view/post (let's say 0.01 ETH)
1. Have another Status account with 2 wallet accounts, one with 0.01 ETH and the other has nothing
2. With that account, go see the channel that has a permission -> you should have access
3. Go in the Edit Shared Addresses popup
4. Unselect the account that has 0.01 ETH

Result: the popup edit correctly showing that you no longer have access to the channel, but in the background, the channel also loses access, even if we didn't send yet.

#### Expected behavior

Only the popup updates until the addresses are actually sent

#### Actual behavior

The chat_section module reacts to the check permission event even if it wasn't meant for it, resulting in the chat content to lock up before the addresses are even sent.

### Additional Information

- Status desktop version: master (`277dda7533c8602f2fe0a4c72615be05ff9df118`)

### PR DESCRIPTION
**Pull request by MishkaRogachev**
Close #14376

### What does the PR do

https://github.com/status-im/status-desktop/assets/2522130/e4d42e95-8d87-4221-822a-75fb8d39590d

- [x] Add enum request id to track permissions check role
- [x] Add dedicated permissions model for addresses check popup

### Affected areas

Communities

### CODE REVIEW

**Comment by cammellos** [comment]
@MishkaRogachev JSON rpc should already have a request id you can use:

```
curl -H "Content-Type: application/json" -X POST --data '{"jsonrpc":"2.0","method":"web3_clientVersion","params":[],"id":67}'
```
can we leverage that instead?

**Comment by MishkaRogachev** [comment]
> @MishkaRogachev JSON rpc should already have a request id you can use:
> 
> ```
> curl -H "Content-Type: application/json" -X POST --data '{"jsonrpc":"2.0","method":"web3_clientVersion","params":[],"id":67}'
> ```
> 
> can we leverage that instead?

you are right in case we need to identify requests between status-go and desktop, should use id from json, here the problem is a bit deeper on ui, i think when i come back to this PR i need to create a second permissions model instance to check permissions independently from the main community permissions model

**Comment by MishkaRogachev** [comment]
@caybro @mprakhov 
I am putting this PR for review, the problem described in the task it fixes, but there are a few buts:
- The components are very interconnected, but I will not risk to do some serious refactoring myself, there are a lot of components whose purpose I don't understand very well.
- There is a problem with the state of the button, the state as if lags by one step. (UPD: the same problem exists in the master):
Screenshot 2024-04-29 at 13 08 20
Screenshot 2024-04-29 at 13 06 33

**Comment by MishkaRogachev** [comment]
@caybro i can't reproduce this issue on master. Is not it already fixed via https://github.com/status-im/status-desktop/pull/14518?

File: src/app/modules/main/chat_section/io_interface.nim [hunk 1]
```diff
@@ -378,7 +378,7 @@ method onAcceptRequestToJoinFailedNoPermission*(self: AccessInterface, community
 method onDeactivateChatLoader*(self: AccessInterface, chatId: string) {.base.} =
   raise newException(ValueError, "No implementation available")
 
-method onCommunityCheckPermissionsToJoinResponse*(self: AccessInterface, checkPermissionsToJoinResponse: CheckPermissionsToJoinResponseDto) {.base.} =
+method onCommunityCheckPermissionsToJoinResponse*(self: AccessInterface, checkPermissionsToJoinResponse: CheckPermissionsToJoinResponseDto, ) {.base.} =
```
[the remark this answers is not in the document]
Comment (MishkaRogachev) [reply src/app/modules/main/chat_section/io_interface.nim]: fixed

**Comment by MishkaRogachev** [comment]
@jrainville @mprakhov @caybro 
I have some doubts that this PR is worth it:
- it doesn't make the code cleaner, the separation should be made deeper
- the problem mentioned in the ticket has already been fixed in the master, or rather I can't reproduce it anymore

If anyone can confirm that the problem is not reproducible in the current master, I suggest closing this PR

**Comment by jrainville** [comment]
> If anyone can confirm that the problem is not reproducible in the current master, I suggest closing this PR

If indeed the issue isn't reproducible anymore, it's not worth adding code for no reason. There might be a refactor on the permissions module later anyway

**Comment by MishkaRogachev** [comment]
> > If anyone can confirm that the problem is not reproducible in the current master, I suggest closing this PR
> 
> If indeed the issue isn't reproducible anymore, it's not worth adding code for no reason. There might be a refactor on the permissions module later anyway

Yeah, I'd prefer a more consistent refactoring here

**Pull request closed without merging**